### PR TITLE
Added a Makefile-Engine Makefile to build the library

### PR DESCRIPTION
**Pull request by jfjhh**
Before, there was no way to build the library at all. The additions allow the library to be created for haiku, using the native `makefile-engine`.

### CODE REVIEW

**Comment by TimothyGu** [comment]
It should be configure.ac.

File: configure.in [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT(AVLDupTree, 1.0.0, https://github.com/HaikuArchives/AVLDupTree/issues)
```
Comment (TimothyGu) [inline configure.in]: I would quote the URL, just to be safe.

File: configure.in [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+AC_CONFIG_SRCDIR([Source/AVLDupTree.c])
+# AC_CONFIG_HEADERS([config.h])
```
Comment (TimothyGu) [inline configure.in]: Please simply remove this.

File: configure.in [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+AC_PROG_CXX
+AC_PROG_CC
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([OS.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
```
Comment (TimothyGu) [inline configure.in]: Is this really necessary?

File: configure.in [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT(AVLDupTree, 1.0.0, https://github.com/HaikuArchives/AVLDupTree/issues)
+AC_CONFIG_SRCDIR([Source/AVLDupTree.c])
+# AC_CONFIG_HEADERS([config.h])
+
+# Checks for programs.
+AC_PROG_CXX
```
Comment (TimothyGu) [inline configure.in]: AVLDupTree isn't an C++ project and doesn't use the C++ compiler. (CXX == C++ compiler)

Comment (TimothyGu) [inline configure.in]: Scratch this.

File: configure.in [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+AC_PROG_CC
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([OS.h stdlib.h string.h])
```
Comment (TimothyGu) [inline configure.in]: You should probably make configure fail if one of these is not available (e.g. on Linux).

**Comment by TimothyGu** [comment]
Actually I'm not sure if this set up would work at all. You at least need to:
1. Set some output files in configure.ac
2. Move Makefile to Makefile.in
3. Edit Makefile.in to get rid of the `CC` variable.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,511 @@
...
+#	Specify the level of optimization that you want. Specify either NONE (O0),
+#	SOME (O1), FULL (O2), or leave blank (for the default optimization level).
+OPTIMIZE := FULL
+
+# 	Specify the codes for languages you are going to support in this
+# 	application. The default "en" one must be provided too. "make catkeys"
+# 	will recreate only the "locales/en.catkeys" file. Use it as a template
+# 	for creating catkeys for other languages. All localization files must be
+# 	placed in the "locales" subdirectory.
+LOCALES = en
```
Comment (waddlesplash) [inline Makefile]: Are there translations in this app? If no, then it doesn't need this.

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+#		DRIVER_PATH = video/usb
+#	will instruct the "driverinstall" rule to place a symlink to your driver's
+#	binary in ~/add-ons/kernel/drivers/dev/video/usb, so that your driver will
+#	appear at /dev/video/usb when loaded. The default is "misc".
+DRIVER_PATH = 
+
+## Include the Makefile-Engine
+# DEVEL_DIRECTORY := \
+#	$(shell findpaths -r "makefile_engine" B_FIND_PATH_DEVELOP_DIRECTORY)
+include $(DEVEL_DIRECTORY)/etc/makefile-engine
```
Comment (waddlesplash) [inline Makefile:128]: Delete this line and uncomment the two lines above it, then delete all the lines below this.

File: configure.ac [hunk 8]
```diff
@@ -0,0 +1,19 @@
+#                                               -*- Autoconf -*-
```
Comment (waddlesplash) [inline configure.ac]: Why are you using autoconf?

Comment (jfjhh) [inline configure.ac]: What else would you recommend?

Comment (waddlesplash) [inline configure.ac]: Just a Makefile-Engine makefile. The code is BeOS/Haiku only anyway.

**Comment by jfjhh** [comment]
At the moment, the [haikuporter recipe file](https://bitbucket.org/haikuports/haikuports/pull-request/452/added-recipe-for-libavlduptree/diff) will fail because without the PR being accepted, the recipe will be unable to find any `Makefile`.

**Comment by waddlesplash** [comment]
In the future, please squash your commits. There is no reason to have 7 commits here for one change, with lots of erroneous stuff in the early commits and corrections in the later ones.

**Pull request merged**